astropy/io/fits/convenience.py

```diff
@@ -709,6 +710,105 @@ def info(filename, output=None, **kwargs):
     return ret
 
 
+def printdiff(inputa, inputb, *args, **kwargs):
+    """
+    Compare two parts of a fits file, including entire FITS files, FITS HDULists and
+    FITS HDU objects.
+
+    Parameters
+    ----------
+    inputa : str of file path, HDUList object, or HDU object
+        The filename of a FITS file on disk, or HDUList or HDU
```

Comment:
"on disk" might be inaccurate here? AFAIK, FITS I/O also supports remote data; although in that case, it does download it first on disk in the cache, but specifying "on disk" explicitly here implies that remote data is unsupported.